### PR TITLE
feat: support override default html-plugin

### PR DESCRIPTION
**Pull request by 9aoy**
## Summary

Some third-party plugins strongly depend on html-webpack-plugin, like webpack-subresource-integrity.

since rspack compat html-webpack-plugin, we can support setHtmlPlugin method to override default html-plugin.

## Checklist

- [ ] Tests updated.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by 9aoy** [comment]
!canary

**Comment by 9aoy** [comment]
!canary

**Comment by 9aoy** [comment]
!canary

**Comment by 9aoy** [comment]
!canary

File: packages/shared/src/types/rsbuild.ts [hunk 1]
```diff
@@ -59,4 +59,10 @@ export type RsbuildInstance<P extends RsbuildProvider = RsbuildProvider> = {
   >['pluginAPI']['onAfterStartProdServer'];
   onDevCompileDone: Awaited<ReturnType<P>>['pluginAPI']['onDevCompileDone'];
   onExit: Awaited<ReturnType<P>>['pluginAPI']['onExit'];
+
+  /**
+   * This method is used to override the Rsbuild default html-plugin (html-rspack-plugin).
+   *
+   */
+  setHtmlPlugin: (plugin: typeof import('html-webpack-plugin')) => void;
```
Comment (chenjiahan) [inline packages/shared/src/types/rsbuild.ts:67]: this is a private helper, we can just export this method from `@rsbuild/core/provider`, so Rsbuild users will not use this method.

File: packages/core/src/provider/htmlPluginUtil.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * This file is used to provide/set a global html-plugin singleton
+ */
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+
+let htmlPlugin: typeof HtmlWebpackPlugin;
+
+/**
+ * This method is used to override the Rsbuild default html-plugin (html-rspack-plugin).
+ */
+export const setHtmlPlugin = (plugin: typeof HtmlWebpackPlugin) => {
```
Comment (chenjiahan) [inline packages/core/src/provider/htmlPluginUtil.ts:11]: setHTMLPlugin / getHTMLPlugin

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**